packages/react/src/components/GroupedList/GroupedListV2.base.tsx

```diff
@@ -0,0 +1,507 @@
+    }
+    events.current = new EventGroup(this);
+    if (selection) {
+      events.current.on(selection, SELECTION_CHANGE, onSelectionChange);
+    }
+  });
+
+  const onSelectionChange = React.useCallback(() => {
+    setVersion({});
+  }, [setVersion]);
+
+  useUnmount(() => {
+    events.current?.dispose();
+  });
+
+  React.useEffect(() => {
+    setVersion({});
+  }, [compact]);
+
+  React.useEffect(() => {
+    const newIsSomeGroupExpanded = computeIsSomeGroupExpanded(groups);
+    if (newIsSomeGroupExpanded !== isSomeGroupExpanded.current) {
+      isSomeGroupExpanded.current = newIsSomeGroupExpanded;
+      onGroupExpandStateChanged?.(newIsSomeGroupExpanded);
+    }
+  }, [groups, toggleVersion, onGroupExpandStateChanged, groupExpandedVersion]);
+
+  const onToggleCollapse = React.useCallback(
+    (group: IGroup): void => {
+      const onToggleCollapseFn = groupProps?.headerProps?.onToggleCollapse;
+
+      if (group) {
+        onToggleCollapseFn?.(group);
+        group.isCollapsed = !group.isCollapsed;
+        setToggleVersion({});
+        setVersion({});
+      }
+    },
+    [setToggleVersion, groupProps],
+  );
+
+  const onToggleSelectGroup = React.useCallback(
+    (group: IGroup): void => {
+      if (group && selection && selectionMode === SelectionMode.multiple) {
+        selection.toggleRangeSelected(group.startIndex, group.count);
+      }
+    },
+    [selection, selectionMode],
+  );
+
+  const onToggleSummarize = React.useCallback(
+    (group: IGroup): void => {
+      const onToggleSummarizeFn = groupProps?.showAllProps?.onToggleSummarize;
+
+      if (onToggleSummarizeFn) {
+        onToggleSummarizeFn(group);
+      } else {
+        if (group) {
+          group.isShowingAll = !group.isShowingAll;
+        }
+
+        setVersion({});
+        setToggleVersion({});
+      }
+    },
+    [groupProps],
+  );
+
+  const renderItem = React.useCallback(
+    (item: IGroupedItem, flattenedIndex: number): React.ReactNode => {
+      const group = item.group;
+
+      const level = group?.level ? group.level + 1 : 1;
+      const isSelected = selection && group ? selection.isRangeSelected(group.startIndex, group.count) : false;
+
+      const dividerProps = {
+        group,
+        groupIndex: flattenedIndex,
+        groupLevel: group?.level ?? 0,
+        isSelected,
+        selected: isSelected,
+        viewport,
+        selectionMode,
+        groups,
+        compact,
+        onToggleSelectGroup,
+        onToggleCollapse,
+        onToggleSummarize,
+      };
+
+      if (item.type === 'group') {
+        const groupHeaderProps = {
+          ...groupProps!.headerProps,
+          ...dividerProps,
+          key: group!.key,
+          groupedListId: item.groupId!,
+          ariaLevel: level,
+          ariaSetSize: groups ? groups.length : undefined,
+          ariaPosInSet: flattenedIndex !== undefined ? flattenedIndex + 1 : undefined,
```

Comment:
The `aria-posinset` value is actually the index within the current level-specific group, not the index within all rows.

Reply:
The reaaaaaally annoying thing is that if this is within a Grouped DetailsList, the index is actually the `flattenedIndex`, and the attribute is `aria-rowindex` (and there's no `aria-level` or `aria-setsize`) 😂😭

Reply:
Made a follow up issue to address: #24795 

> and there's no aria-level or aria-setsize

We should probably remove them from the existing GroupedList implementation as well then :D